Products/ZenUtils/extdirect/router.py

```diff
@@ -174,8 +178,15 @@ def _processDirectRequest(self, directRequest):
         if isinstance(data, (int, basestring)):
             data = {'id': data}
 
+        # Set default value for event class for transform getting without chosen event class
+        if not data:
+            if action == 'EventClassesRouter' and (method == 'getTransformTree' or method == 'isTransformEnabled'):
+                data = {'uid': 'zport/dmd/Events'}
+            elif action == 'DetailNavRouter' and method == 'getSecurityPermissions':
+                data = {'uid': 'zport/dmd/Events'}
```

Comment:
Rather using conditionals here to correct issues elsewhere, why not fix the issue where it originates?  The methods `getTransformTree`, `isTransformEnabled`, and `getSecurityPermissions` are each called from one place in the Javascript code.  And they're all called from within a `setContext` function that passes in the `uid` value, so something else is calling `setContext` and passing in no value for `uid`.

We should try to fix the issue at where the problem originates.

Also, `DetailNavRouter` seems to be a general purpose component and not specific to the Event Console. I'm not convinced that setting `uid` to the value "zport/dmd/Events" would be correct in all situations.

Reply:
The solution was improved according to the remarks and suggestions noted above.